### ISSUE TITLE
porter build fails when connecting to docker daemon via TCP+TLS

### ISSUE THREAD

**Issue by andreas-baus** [getporter/porter#1532]
## Describe the bug
We have set up a build pipeline on a custom kubernetes-based GitLab runner where we were unable to use the default socket-based connection to the docker daemon, so we had to use tcp. With this setup, 'docker build' commands still work; but 'porter build' commands fail during the 'Invocation Image Build' step because the connection to the docker daemon fails.

## To Reproduce
1. Set up docker to use TLS
2. In particular, set the DOCKER_HOST environment variable to something like DOCKER_HOST=tcp://localhost:2376
3. Run 'docker build' with a minimal Dockerfile (like the one from [this example](https://riptutorial.com/docker/example/10772/helloworld-dockerfile)) - this should succeed
4. Run `porter build` with a minimal porter bundle (like the Hello World example from the porter [quickstart guide](https://porter.sh/quickstart/))
5. This will fail due to connection errors when the docker invoked by the porter runtime tries to talk to the docker daemon

## Expected behavior
The `docker build` spawned by the porter runtime should succeed if a `docker build` run directly in the same environment succeeds.

## Porter Command and Output
```
$ porter build --verbose --debug
Running linters for each mixin used in the manifest...
DEBUG name:    arm
DEBUG pkgDir: /root/.porter/mixins/arm
DEBUG file:     
DEBUG stdin:
/builds/ddgix/test/porter-test/cnab/root/.porter/mixins/arm/arm version --output json --debug
DEBUG name:    aws
DEBUG pkgDir: /root/.porter/mixins/aws
DEBUG file:     
DEBUG stdin:
/builds/ddgix/test/porter-test/cnab/root/.porter/mixins/aws/aws version --output json --debug
DEBUG name:    az
DEBUG pkgDir: /root/.porter/mixins/az
DEBUG file:     
DEBUG stdin:
/builds/ddgix/test/porter-test/cnab/root/.porter/mixins/az/az version --output json --debug
DEBUG name:    exec
DEBUG pkgDir: /root/.porter/mixins/exec
DEBUG file:     
DEBUG stdin:
/builds/ddgix/test/porter-test/cnab/root/.porter/mixins/exec/exec version --output json --debug
DEBUG name:    gcloud
DEBUG pkgDir: /root/.porter/mixins/gcloud
DEBUG file:     
DEBUG stdin:
/builds/ddgix/test/porter-test/cnab/root/.porter/mixins/gcloud/gcloud version --output json --debug
DEBUG name:    helm
DEBUG pkgDir: /root/.porter/mixins/helm
DEBUG file:     
DEBUG stdin:
/builds/ddgix/test/porter-test/cnab/root/.porter/mixins/helm/helm version --output json --debug
DEBUG name:    helm3-centros
DEBUG pkgDir: /root/.porter/mixins/helm3-centros
DEBUG file:     
DEBUG stdin:
/builds/ddgix/test/porter-test/cnab/root/.porter/mixins/helm3-centros/helm3-centros version --output json --debug
DEBUG name:    kubernetes
DEBUG pkgDir: /root/.porter/mixins/kubernetes
DEBUG file:     
DEBUG stdin:
/builds/ddgix/test/porter-test/cnab/root/.porter/mixins/kubernetes/kubernetes version --output json --debug
DEBUG name:    terraform
DEBUG pkgDir: /root/.porter/mixins/terraform
DEBUG file:     
DEBUG stdin:
/builds/ddgix/test/porter-test/cnab/root/.porter/mixins/terraform/terraform version --output json --debug
Copying porter runtime ===> 
Copying mixins ===> 
Copying mixin exec ===> 
Generating Dockerfile =======>
DEBUG name:    exec
DEBUG pkgDir: /root/.porter/mixins/exec
DEBUG file:     
DEBUG stdin:
actions:
  install:
  - exec:
      arguments:
      - install
      command: ./helpers.sh
      description: Install Hello World
  uninstall:
  - exec:
      arguments:
      - uninstall
      command: ./helpers.sh
      description: Uninstall Hello World
  upgrade:
  - exec:
      arguments:
      - upgrade
      command: ./helpers.sh
      description: World 2.0
/builds/ddgix/test/porter-test/cnab/root/.porter/mixins/exec/exec build --debug
FROM debian:stretch-slim
ARG BUNDLE_DIR
RUN apt-get update && apt-get install -y ca-certificates
# exec mixin has no buildtime dependencies
# Use the BUNDLE_DIR build argument to copy files into the bundle
COPY . $BUNDLE_DIR
RUN rm $BUNDLE_DIR/porter.yaml
RUN rm -fr $BUNDLE_DIR/.cnab
COPY .cnab /cnab
WORKDIR $BUNDLE_DIR
CMD ["/cnab/app/run"]
Writing Dockerfile =======>
FROM debian:stretch-slim
ARG BUNDLE_DIR
RUN apt-get update && apt-get install -y ca-certificates
# exec mixin has no buildtime dependencies
# Use the BUNDLE_DIR build argument to copy files into the bundle
COPY . $BUNDLE_DIR
RUN rm $BUNDLE_DIR/porter.yaml
RUN rm -fr $BUNDLE_DIR/.cnab
COPY .cnab /cnab
WORKDIR $BUNDLE_DIR
CMD ["/cnab/app/run"]
Starting Invocation Image Build (core.harbor.sandbox4.centros.cloud/test/porter-hello-installer:v0.1.0) =======> 
time="2021-03-30T14:19:20Z" level=error msg="Can't add file /builds/ddgix/test/porter-test/cnab/.cnab/app/mixins/exec/exec to tar: io: read/write on closed pipe"
time="2021-03-30T14:19:20Z" level=error msg="Can't close tar writer: io: read/write on closed pipe"
Error: unable to build CNAB invocation image: error during connect: Post http://docker:2376/v1.24/build?buildargs=%7B%22BUNDLE_DIR%22%3A%22%2Fcnab%2Fapp%22%7D&cachefrom=null&cgroupparent=&cpuperiod=0&cpuquota=0&cpusetcpus=&cpusetmems=&cpushares=0&dockerfile=Dockerfile&labels=null&memory=0&memswap=0&networkmode=&rm=1&shmsize=0&t=core.harbor.sandbox4.centros.cloud%2Ftest%2Fporter-hello-installer%3Av0.1.0&target=&ulimits=null&version=: write tcp 127.0.0.1:50212->127.0.0.1:2376: write: broken pipe
```

## Version
porter v0.33.1 (4359277d)

**Comment by VinozzZ** [comment]
I followed the setup in the docker doc for my local docker daemon https://docs.docker.com/engine/security/protect-access/#use-tls-https-to-protect-the-docker-daemon-socket. 
I used the porter `v1.0.0-alpha.20`.
`docker build` was able to successfully built an image 
I was able to reproduce the error using `porter build`
```
Using buildkit build driver
Running linters for each mixin used in the manifest...
DEBUG name:    exec
DEBUG pkgDir: /home/yingrong/.porter/mixins/exec
DEBUG file:     
DEBUG stdin:

/home/yingrong/work/example-bundles/hello/home/yingrong/.porter/mixins/exec/exec version --output json --debug
Defaulting the type of output name to file
Copying porter runtime ===> 
Copying mixins ===> 
Copying mixin exec ===> 

Generating Dockerfile =======>
DEBUG name:    exec
DEBUG pkgDir: /home/yingrong/.porter/mixins/exec
DEBUG file:     
DEBUG stdin:
actions:
  install:
    - exec:
        arguments:
          - install
        command: ./helpers.sh
        description: Install Hello World
  uninstall:
    - exec:
        arguments:
          - uninstall
        command: ./helpers.sh
        description: Uninstall Hello World
  upgrade:
    - exec:
        arguments:
          - upgrade
        command: ./helpers.sh
        description: World 2.0

/home/yingrong/work/example-bundles/hello/home/yingrong/.porter/mixins/exec/exec build --debug
# syntax=docker/dockerfile-upstream:1.4.0
FROM debian:stretch-slim

ARG BUNDLE_DIR
ARG BUNDLE_UID=65532
ARG BUNDLE_USER=nonroot
ARG BUNDLE_GID=0
RUN useradd ${BUNDLE_USER} -m -u ${BUNDLE_UID} -g ${BUNDLE_GID} -o

RUN rm -f /etc/apt/apt.conf.d/docker-clean; echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache
RUN --mount=type=cache,target=/var/cache/apt --mount=type=cache,target=/var/lib/apt \
    apt-get update && apt-get install -y ca-certificates

# exec mixin has no buildtime dependencies


COPY --link . ${BUNDLE_DIR}
RUN rm ${BUNDLE_DIR}/porter.yaml
RUN rm -fr ${BUNDLE_DIR}/.cnab
COPY --link .cnab /cnab
RUN chgrp -R ${BUNDLE_GID} /cnab && chmod -R g=u /cnab
USER ${BUNDLE_UID}
WORKDIR ${BUNDLE_DIR}
CMD ["/cnab/app/run"]

Writing Dockerfile =======>
# syntax=docker/dockerfile-upstream:1.4.0
FROM debian:stretch-slim

ARG BUNDLE_DIR
ARG BUNDLE_UID=65532
ARG BUNDLE_USER=nonroot
ARG BUNDLE_GID=0
RUN useradd ${BUNDLE_USER} -m -u ${BUNDLE_UID} -g ${BUNDLE_GID} -o

RUN rm -f /etc/apt/apt.conf.d/docker-clean; echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache
RUN --mount=type=cache,target=/var/cache/apt --mount=type=cache,target=/var/lib/apt \
    apt-get update && apt-get install -y ca-certificates

# exec mixin has no buildtime dependencies


COPY --link . ${BUNDLE_DIR}
RUN rm ${BUNDLE_DIR}/porter.yaml
RUN rm -fr ${BUNDLE_DIR}/.cnab
COPY --link .cnab /cnab
RUN chgrp -R ${BUNDLE_GID} /cnab && chmod -R g=u /cnab
USER ${BUNDLE_UID}
WORKDIR ${BUNDLE_DIR}
CMD ["/cnab/app/run"]
Building invocation image
[+] Building 0.0s (0/0)                                                                                                        
error building docker image: Error response from daemon: Client sent an HTTP request to an HTTPS server.: driver not connecting
unable to build CNAB invocation image: error building docker image: Error response from daemon: Client sent an HTTP request to an HTTPS server.: driver not connecting
```

**Comment by carolynvs** [comment]
@VinozzZ Which environment variables are you setting that got docker build to work? I can dig around in the code to see what we are missing to pick them up.

**Comment by VinozzZ** [comment]
` export DOCKER_HOST=tcp://$HOST:2376 DOCKER_TLS_VERIFY=1`
it's these two

**Comment by carolynvs** [comment]
Ah okay I think I see why we aren't getting those for free. The TLS options are being set in the cli flag setup, and there isn't a reusable function that we can use to get similar functionality. I'll check env vars and manually populate the TLS options in the Porter driver itself.

**Comment by carolynvs** [comment]
@andreas-baus This is now available in Porter [v1.0.0-alpha.21](https://github.com/getporter/porter/releases/tag/v1.0.0-alpha.21)
